datafusion/common/src/file_options/csv_writer.rs

```diff
@@ -91,6 +91,11 @@ impl TryFrom<(&ConfigOptions, &StatementOptions)> for CsvWriterOptions {
                         )
                     })?)
             },
+                "quote" | "escape" => {
+                    // These two attributes are only available when reading csv files.
```

Comment:
Yeah, this is an gap in the arrow CSV handling. I filed https://github.com/apache/arrow-rs/issues/5146

```suggestion
                    // https://github.com/apache/arrow-rs/issues/5146
                    // These two attributes are only available when reading csv files.
```